src/mds/Server.cc

```diff
@@ -362,8 +362,19 @@ void Server::dispatch(const cref_t<Message> &m)
     handle_peer_request(ref_cast<MMDSPeerRequest>(m));
     return;
   default:
-    derr << "server unknown message " << m->get_type() << dendl;
-    ceph_abort_msg("server unknown message");  
+    if(m->get_connection()->get_peer_type() == CEPH_ENTITY_TYPE_MDS) {
+      derr << "MDS peer sending unknown message to MDS" << m->get_type() << dendl;
+      ceph_abort_msg("Aborting: MDS peer sending unknown message");
+    } else if(m->get_connection()->get_peer_type() == CEPH_ENTITY_TYPE_CLIENT) {
+      const auto &req = ref_cast<MClientRequest>(m);
+      Session *session = mds->get_session(req);
+      derr << "server unknown message: " << m->get_type() << ", therefore closing session and blocklisting client " << session->get_client() << dendl;
+      CachedStackStringStream css;
+      mds->evict_client(session->get_client().v, false, true, *css, nullptr);
+    } else {
+      derr << "server unknown message " << m->get_type() << dendl;
+      ceph_abort_msg("Aborting: server unknown message");
+    }
```

Comment:
Since you are printing the message type, IMO you can also always print the peer type in else and then :

```
if(m->get_connection()->get_peer_type() == CEPH_ENTITY_TYPE_CLIENT) {
} else { // for all the other types including the MDS
 // print messge type and peer type.
}
```

Reply:
Right, Thanks for the comment. Made the changes and pushed.